templates/OperationSnippet.stencil

```diff
@@ -10,18 +10,42 @@ public func {{ op.name }}(
     guard let url = self.url(forTemplate: urlTemplate) else { return }
 
     // Construct query
+    {% if op.queryInitParams.count > 0 %}
     var queryParams: [QueryParameter] = [
-        {% for param in op.queryParams %}
-        "{{ param.key }}" : {{ param.value}}
+        {% for param in op.queryInitParams %}
+        ("{{ param.key }}" , {{ param.value}})
         {% endfor %}
     ]
+    {% elif op.queryParams.count > 0 %}
+    var queryParams = [QueryParameter]()
+    {% endif %}
+
+    {% if op.queryParams.count > 0 %}
+    {% for param in op.queryParams %}
+    if {{ param.value }} != nil {
+        queryParams.append("{{ param.key }}", {{ param.value}})
+    }
+    {% endfor %}
+    {% endif %}
```

Comment:
There's no need for this if statement if `op.queryParams` has a default value of empty collection.